### ISSUE TITLE
Component doesn't generate Module

### ISSUE THREAD

**Issue by kensodemann** [ionic-team/ionic-cli#2632]
_From @Sampath-Lokuge on August 14, 2017 3:18_

[X ] **3.x**

[X ] feature request

**Current behavior:**
`ionic generate component SubscribeTopicComponent` CLI command doesn't generate the `Module` file.

**Expected behavior:**
We need a `Module` by default like `Page` component. But you can give `--no-module` for someone who doesn't like this feature.In other words, we like the same way of generation `Page` CLI commands. 

**Ionic info:**

> @ionic/cli-plugin-cordova       : 1.6.2
>     @ionic/cli-plugin-ionic-angular : 1.4.1
>     @ionic/cli-utils                : 1.7.0
>     ionic (Ionic CLI)               : 3.7.0
> 
> global packages:
> 
>     Cordova CLI : 7.0.1
> 
> local packages:
> 
>     @ionic/app-scripts : 2.1.3
>     Cordova Platforms  : android 6.2.3
>     Ionic Framework    : ionic-angular 3.6.0
> 
> System:
> 
>     Android SDK Tools : 25.2.5
>     Node              : v6.10.0
>     OS                : Windows 8.1
>     npm               : 3.10.10

_Copied from original issue: ionic-team/ionic#12658_

**Comment by dwieeb** [comment]
Thanks @kensodemann 

@Sampath-Lokuge For me it seems to be making/modifying a `src/components/components.module.ts` file, which looks like a shared NgModule for components. 

@mhartington Is this the correct behavior?

**Comment by Sampath-Lokuge** [comment]
@dwieeb Actually I would like to have my component wise module for each component.Not as a shared module.This feature was there in the previous versions.But now it is not there.

Use case: When I need to use that component I can easily import it into relevant page's module.I would like to use component per module approach rather than shared module approach.

**Comment by dwieeb** [comment]
@Sampath-Lokuge I understand, I was merely posting my findings, as this behavior is actually new to me. 😄

**Comment by Sampath-Lokuge** [comment]
@dwieeb OK sure 👍 

Here you can see the [**official blog**](http://blog.ionic.io/ionic-and-lazy-loading-pt-2/) which I have followed to do above approach.

![Image](http://i.imgur.com/USGcG4N.png)

**Comment by Sampath-Lokuge** [comment]
@dwieeb And now component doesn't generate `@Input()` and `@Output()` meta tags too.We need to add it manually.If I can remember properly those were there in the earlier versions too.

**Comment by Sampath-Lokuge** [comment]
#2635

**Comment by mhartington** [comment]
Creating shared modules was the solution we went with for the generators. It offers the more flexible solution for larger apps and keeps the number of ngModules down to a minimum.

@Sampath-Lokuge the blog post you have linked to just points out 2 different options. With the generators, I went with option 2.

It could be possible to add support for this down the road, but it's not an immediate priority.
